ikasaneip/rest/rest-dashboard/src/main/java/org/ikasan/rest/dashboard/MetaDataApplication.java

```diff
@@ -26,27 +41,84 @@
+        }
+
         this.mapper = new ObjectMapper();
+        this.mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+        SimpleModule m = new SimpleModule();
+        m.addAbstractTypeMapping(ModuleMetaData.class, ModuleMetaDataImpl.class);
+        m.addAbstractTypeMapping(FlowMetaData.class, FlowMetaDataImpl.class);
+        m.addAbstractTypeMapping(FlowElementMetaData.class, FlowElementMetaDataImpl.class);
+        m.addAbstractTypeMapping(Transition.class, TransitionImpl.class);
+        m.addAbstractTypeMapping(ConfigurationParameterMetaData.class,ConfigurationParameterMetaDataImpl.class);
+        m.addAbstractTypeMapping(ConfigurationMetaData.class,ConfigurationMetaDataImpl.class);
+
+        this.mapper.registerModule(m);
     }
 
     @RequestMapping(method = RequestMethod.PUT,
         value = "/module/metadata")
     @PreAuthorize("hasAnyAuthority('ALL','WebServiceAdmin')")
-    public ResponseEntity harvestWiretap(@RequestBody String metadataJsonPayload)
+    public ResponseEntity harvestModuleMetadata(@RequestBody String moduleMetadataJsonPayload)
     {
-//        try
-//        {
-            logger.info(metadataJsonPayload);
-//        }
-//        catch (IOException e)
-//        {
-//            e.printStackTrace();
-//            throw new ResponseStatusException(
-//                HttpStatus.BAD_REQUEST, "Cannot parse wiretap JSON!", e);
-//        }
+        try
+        {
+            logger.debug(moduleMetadataJsonPayload);
+            ModuleMetaData moduleMetadata = this.mapper.readValue(moduleMetadataJsonPayload
+                , ModuleMetaDataImpl.class);
+
+            List<ModuleMetaData> entities = new ArrayList<>();
+            entities.add(moduleMetadata);
+
+            this.moduleMetaDataBatchInsert.insert(entities);
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST
+                , "An error has occurred attempting to perform a batch insert of ModuleMetaData!", e);
+        }
 
         return new ResponseEntity("Module metadata successfully captured!", HttpStatus.OK);
     }
+
+    @RequestMapping(method = RequestMethod.PUT,
+        value = "/configuration/metadata")
+    @PreAuthorize("hasAnyAuthority('ALL','WebServiceAdmin')")
+    public ResponseEntity harvestConfigurationMetadata(@RequestBody String configurationMetadataJsonPayload)
+    {
+        try
+        {
+            logger.debug(configurationMetadataJsonPayload);
+
+            List<ConfigurationMetaData> configurationMetaDataList = this.mapper.readValue(configurationMetadataJsonPayload
+                , mapper.getTypeFactory().constructCollectionType(List.class, ConfigurationMetaDataImpl.class));
+
+            this.configurationMetaDataBatchInsert.insert(configurationMetaDataList);
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
```

Comment:
there is no need to print stacktrace it is being passed to RsponseStatusException

Reply:
Removed all stack traces.